code/go/0chain.net/smartcontract/storagesc/validator_test.go

```diff
@@ -0,0 +1,30 @@
+package storagesc
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidatorHealthCheck(t *testing.T) {
+
+	var (
+		ssc      = newTestStorageSC()
+		balances = newTestBalances(t, false)
+
+		tp int64 = 100
+	)
+
+	setConfig(t, balances)
+
+	var (
+		validator = addValidator(t, ssc, tp, balances)
+		v, err    = getValidator(validator.id, balances)
+	)
+
+	require.NoError(t, err)
+
+	_, err = healthCheckValidator(t, v, 0, tp, ssc, balances)
+
+	require.NoError(t, err)
```

Comment:
check that data is really updated

Reply:
They are getting checked in healthCheckBlobber / healthCheckValidator as the transaction are created in these functions and we need to check with tx.CreationData